src/vmm_mad/remotes/lib/vcenter_driver/virtual_machine.rb

```diff
@@ -1303,6 +1438,8 @@ def calculate_add_nic_spec(nic)
                  :port => port)
         end
 
+        # grab the last unitNumber to ensure the nic to be added at the end
+        @unic = @unic || get_vcenter_nics.map{|d| d.unitNumber}.max rescue 0
```

Comment:
maybe we need this to fail with an error message instead of 'rescue 0'?

Reply:
rescue 0 means that get_vcenter_nics array is empty, so we are rescuing this with the 0 unitNumber